mne/decoding/tests/test_transformer.py

```diff
@@ -160,3 +164,33 @@ def test_epochs_vectorizer():
     # Test init exception
     assert_raises(ValueError, vector.fit, epochs, y)
     assert_raises(ValueError, vector.transform, epochs, y)
+
+
+@requires_sklearn
+def test_unsupervised_spatial_filter():
+    from sklearn.decomposition import PCA
+    raw = io.read_raw_fif(raw_fname, preload=False)
+    events = read_events(event_name)
+    picks = pick_types(raw.info, meg=True, stim=False, ecg=False,
+                       eog=False, exclude='bads')
+    picks = picks[1:13:3]
+    epochs = Epochs(raw, events, event_id, tmin, tmax, picks=picks,
+                    preload=True, baseline=None, verbose=False)
+
+    # Test fit
+    X = epochs.get_data()
+    usf = UnsupervisedSpatialFilter(PCA(4))
+    usf.fit(X)
+    usf1 = UnsupervisedSpatialFilter(PCA(4))
+
+    # test transform
+    assert_equal(usf.transform(X).ndim, 3)
+    # test fit_transform
+    assert_array_almost_equal(usf.transform(X), usf1.fit_transform(X))
+    # assert shape
+    assert_equal(usf.transform(X).shape[1], 4)
+
+    # Test with average param
+    usf = UnsupervisedSpatialFilter(PCA(4), average=True)
```

Comment:
add test to check average : e.g. `assert_raise(ValueError, UnsupervisedSpatialFilter, PCA(4), average='foo')`